native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/deserializer/DeserializerFactory.cs

```diff
@@ -0,0 +1,42 @@
+using AWSGsrSerDe.common;
+using AWSGsrSerDe.deserializer.avro;
+
+namespace AWSGsrSerDe.deserializer
+{
+    public class DeserializerFactory
+    {
+        private static DeserializerFactory _deserializerFactoryInstance;
+        private AvroDeserializer _avroDeserializer;
+
+        public static DeserializerFactory GetInstance()
+        {
+            return _deserializerFactoryInstance ??= new DeserializerFactory();
+        }
+
+        private DeserializerFactory()
+        {
+        }
+
+        public IDataFormatDeserializer GetDeserializer(string dataFormat, GlueSchemaRegistryConfiguration configs)
+        {
+            switch (dataFormat)
+            {
+                case "AVRO":
```

Comment:
Shouldn't we be using the enum here? Comment applicable everywhere.

Reply:
nice catch, updated.